core/bounds.gms

```diff
@@ -5,120 +5,186 @@
+    vm_deltaCap.fx("2015",regi,te,rlf) = 0;
+    vm_cap.lo("2010",regi,te,rlf) = 0;
+    vm_cap.lo("2015",regi,te,rlf) = 0;
+    vm_cap.lo("2020",regi,te,rlf) = 0;
+  );
 );
 
-vm_demPe.fx(t,regi,"pecoal","seel","pcc") = 0;
-vm_demPe.fx(t,regi,"pecoal","seel","pco") = 0;
-vm_prodSe.fx(t,regi,"pecoal","seel","pcc") = 0;
-vm_prodSe.fx(t,regi,"pecoal","seel","pco") = 0;
-*** windoffshore-todo: to remove when removing wind from all_te
-vm_demPe.fx(t,regi,"pewin","seel","wind") = 0;
-vm_prodSe.fx(t,regi,"pewin","seel","wind") = 0;
+*' no technologies with tech_stat 4 before 2025
+vm_cap.fx(t,regi,te,rlf) $ (t.val <= 2020 and pm_data(regi,"tech_stat",te) = 4) = 0;
+*** initialize cumulative capacity of tech_stat 4 technologies at 0 
+*** (not at ccap0 from generisdata_tech.prn which gives the cucmulative capacity
+***  at the initial investment cost of the first year in which the technology can be built)
+vm_capCum.fx(t0,regi,teLearn) $ (pm_data(regi,"tech_stat",teLearn) = 4) = 0;
+*** tech_stat 4 technologies don't learn before 2025, so capital cost should be fixed
+vm_costTeCapital.fx(t,regi,teLearn) $ (t.val <= 2020 and pm_data(regi,"tech_stat",teLearn) = 4) = fm_dataglob("inco0",teLearn);
 
-*' Switch off coal-h2 hydrogen investments after 2020, and gas-h2 investments after 2030. Our current seh2 hydrogen represents only additional (clean) hydrogen use cases to current ones.
-*' However, as we have too high H2 demand in 2025 and 2030 from the input data, we need to allow grey hydrogen for these time periods to meet the hydrogen demand
-*' which cannot be fully met by incoming low-carbon H2 techologies. This should be removed once FE H2 industry input data is adapted.
-*' It is allowed before 2020 to not make the model infeasible for low demands of hydrogen in that year.
-vm_deltaCap.fx(t,regi,"coalh2",rlf)$(t.val ge 2020) = 0;
-vm_deltaCap.fx(t,regi,"gash2",rlf)$(t.val gt 2030)  = 0;
-vm_cap.lo(t,regi,"coalh2",rlf)$(t.val ge 2020) = 0;
-vm_cap.lo(t,regi,"gash2",rlf)$(t.val gt 2030)  = 0;
+*** no technologies with tech_stat 5 before 2030
+vm_deltaCap.fx(t,regi,te,rlf) $ (t.val <= 2025 and pm_data(regi,"tech_stat",te) = 5) = 0;
 
 
-*' upper bound of 0.5 EJ/yr on grey hydrogen to prevent building too much grey H2 before 2020, distributed to regions via GDP share
-vm_cap.up("2020",regi,"gash2","1") =  0.5 / 3.66 * 1e3 / 8760 * pm_gdp("2020",regi) / sum(regi2,pm_gdp("2020",regi2));
+*** RP: add lower bound on 2020 coal chp and upper bound on gas chp based on IEA data to have a more realistic starting point
+vm_prodSe.lo("2020",regi,"pecoal","seel","coalchp") = 0.8 * pm_IO_output("2020",regi,"pecoal","seel","coalchp") ;
+vm_prodSe.up("2020",regi,"pegas","seel","gaschp") = 1e-4 + 1.3 * pm_IO_output("2020",regi,"pegas","seel","gaschp") ;
 
 
-*' @stop
+*** ------------------------------------------------------------------
+*** ------------------------------------------------------------------
+*' #### Assumptions on biomass
```

Comment:
I would suggest to add a "can be moved to biomass module?" comment here. I also inserted these kinds of comments in the core/declarations restructuring and we now agreed that RSE will start a process to check whether to move these parts.